services/api-server/tests/unit/api_functions/conftest.py

```diff
@@ -0,0 +1,358 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=no-value-for-parameter
+# pylint: disable=super-init-not-called
+# pylint: disable=unused-argument
+# pylint: disable=no-self-use
+
+from uuid import uuid4
+
+import pytest
+from fastapi import FastAPI
+from models_library.api_schemas_webserver.functions_wb_schema import (
+    Function,
+    FunctionIDNotFoundError,
+    FunctionJob,
+    FunctionJobCollection,
+    FunctionJobCollectionIDNotFoundError,
+    FunctionJobIDNotFoundError,
+    RegisteredFunction,
+    RegisteredFunctionJob,
+    RegisteredFunctionJobCollection,
+)
+from models_library.rest_pagination import PageMetaInfoLimitOffset
+from pydantic import TypeAdapter
+from pytest_mock import MockerFixture
+from pytest_simcore.helpers.monkeypatch_envs import setenvs_from_dict
+from pytest_simcore.helpers.typing_env import EnvVarsDict
+from servicelib.rabbitmq._client_rpc import RabbitMQRPCClient
+from simcore_service_api_server.api.routes.functions_routes import get_wb_api_rpc_client
+from simcore_service_api_server.services_rpc.wb_api_server import WbApiRpcClient
+
+
+@pytest.fixture
+def app_environment(
+    app_environment: EnvVarsDict,
+    monkeypatch: pytest.MonkeyPatch,
+):
+    return setenvs_from_dict(
+        monkeypatch,
+        {
+            **app_environment,  # WARNING: AFTER env_devel_dict because HOST are set to 127.0.0.1 in here
+            "WEBSERVER_DEV_FEATURES_ENABLED": "1",
+            "WEBSERVER_FUNCTIONS": "1",
+        },
+    )
+
+
+class DummyRpcClient(RabbitMQRPCClient):
+
+    def __init__(self):
+        self.client_name = "dummy_client"
+        self.settings = {}  # type: ignore # Add a settings attribute to avoid AttributeError
+
+    async def request(self, namespace: str, method_name: str, **kwargs):
+        # Mock implementation of the request method
+        assert isinstance(namespace, str)
+        assert isinstance(method_name, str)
+        assert isinstance(kwargs, dict)
+        return {"mocked_response": True}
+
+
+@pytest.fixture
+async def mock_wb_api_server_rpc(app: FastAPI, mocker: MockerFixture) -> MockerFixture:
+
+    app.dependency_overrides[get_wb_api_rpc_client] = lambda: WbApiRpcClient(
+        _client=DummyRpcClient()
+    )
+    return mocker
+
+
+class MockFunctionRegister:
```

Comment:
ok. so this is used as side-effects for each of the functions of the RPC client API

FYI: We gather all those in one place for each of the simcore services. For instance, in https://github.com/itisfoundation/osparc-simcore/blob/9d6213688ed5124e80bdecdd2cb472b70b86dac7/packages/pytest-simcore/src/pytest_simcore/helpers/webserver_rpc_server.py

- This way we can reuse them elsewhere
- We also are experimenting with more techniques so that we have a single way everywhere to mock the rpc interfaces. Same way that for rest-apis, we use a workflow including https://lundberg.github.io/respx/ 
- I understand this is sufficient here but please have a look at the link above and at least adapt a bit the names e.g. of the mocks etc so it resembles the standard workflow

Reply:
Andrei suggested another way by mocking in place, i got rid of the entire functionregister mock